### PR TITLE
Add module for a cleaner navigation walker

### PR DESCRIPTION
**Pull request by retlehs**
this is the same walker from the sage theme except for:

* bootstrap-specifics have been removed
* make the walker the default if one isn't defined (basically reverting https://github.com/roots/sage/pull/1217)

### CODE REVIEW

**Comment by retlehs** [comment]
@vinkla thanks, but will tackle those outside of this PR.

File: soil.php [hunk 1]
```diff
@@ -50,6 +50,8 @@ public function set($options) {
   }
 }
 
+require_once(plugin_dir_path(__FILE__) . 'lib/utils.php');
```
Comment (QWp6t) [inline soil.php]: `plugin_dir_path` is just a wrapper for `dirname(__FILE__)`

Could use `require_once __DIR__ . '/lib/utils.php;`

**Comment by vinkla** [comment]
@retlehs okay, great. Let me know if I can do anything to help. 

Out of curiosity, what are the standards set for this package?

**Comment by vinkla** [comment]
What about naming the directory `src` instead of `lib`? `src` is much more common than `lib`.

**Comment by swalkinshaw** [comment]
@vinkla coding standards? See here: https://github.com/roots/soil/blob/master/ruleset.xml and more details here https://github.com/roots/guidelines

**Comment by vinkla** [comment]
@swalkinshaw Didn't know about them. Thanks!

**Pull request merged**